docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,218 @@
+# Setup to use SRIO-V with Clear Containers
+
+Single Root IO Virtualization enables splitting a physical device into
+virtual functions (VF) which enables direct pass through to virtual machines or
+containers.  For Clear Containers, we have enabled a CNM plugin and necessary
+changes to facilitate making use of SRIO-V for network based devices.
+
+## Install the SRIO-V Docker\* plugin
+
+To create a network with VFs associated with it which can be passed to
+Clear Containers, install a SRIOV Docker plugin. The created network will be
+based on a physical function (PF) device, and can create 'n' containers, where
+'n' is the number of VFs associated with the PF
+
+To install the plugin, you can follow the [plugin installation instructions](https://github.com/clearcontainers/sriov).
+
+
+## Host setup for SRIO-V
+
+You may need to rebuild your system's kernel in order to disable VFNOIOMMU in
+the kernel config and potentially add a PCI quirk for your NIC. If not, you can
+move to next section. The below section describes how to assess if
+changes are needed, and how to make these changes.
+
+### IOMMU Groups and PCIe Access Control Services
+Taking a look at how the iommu groups are setup on your host system can help
+provide information on whether or not your NIC is setup appropriately with
+respect to PCIe Access Control Services. More specifically, if the PCI bridge is
+within the same iommu-group as your NIC, then this is an indication that either
+your device doesn't support ACS or that it doesn't share this capability
+appropriately by default.
+
+For example, when you run the following command, if all is setup properly, you
+should have the PCI for each ACS enabled NIC port in its own iommu_group. If
+you don't see any output when running the following, then you likely need to
+update your host's kernel config to disable VFNOIOMMU.
+
+```
+find /sys/kernel/iommu_groups/ -type l
+```
+
+For more details, see http://vfio.blogspot.com/2014/08/iommu-groups-inside-and-out.html
+
+### Updating the Host Kernel
+
+These directions are provided as an example, and are based on Ubuntu 16.04. The
+user will need to do equivalent for their distribution.
+
+1.  Grab kernel sources
+
+```
+sudo apt-get install linux-source-4.10.0
+sudo apt-get install linux-headers-4.10.0
+cd /usr/src/linux-source-4.10.0/
+sudo tar -xvf linux-source-4.10.0.tar.bz2
+cd linux-source-4.10.0
+sudo apt-get install libssl-dev
+```
+
+2. Check the config and update if necessary
+
+```
+sudo cp /boot/config-4.8.0-36-generic .config
+sudo make olddefconfig #and verify resulting .config does not have NOIOMMU set; ie: # CONFIG_VFIO_NOIOMMU is not set
```

Comment:
Can you move this comment to a separate line above the command to run. Also, please can you quote the config string so it shows as `"# CONFIG_VFIO_NOIOMMU is not set"`.